homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -108,13 +110,33 @@ def valid_color_configuration(config: ConfigType) -> ConfigType:
     return config
 
 
+def generate_brightness_scale_map(
+    min_default: int, max_default: int, min_device: int, max_device: int
+) -> dict[int, int]:
+    """Generate a mapping from the default brightness scale to the device brightness scale."""
+    mapping = {}
+    for value in range(min_default, max_default + 1):
+        scaled_value = round(
+            (value - min_default)
+            / (max_default - min_default)
+            * (max_device - min_device)
+            + min_device
+        )
+        mapping[scaled_value] = value
+
+    return mapping
+
+
 _PLATFORM_SCHEMA_BASE = (
     MQTT_RW_SCHEMA.extend(
         {
             vol.Optional(CONF_BRIGHTNESS, default=DEFAULT_BRIGHTNESS): cv.boolean,
             vol.Optional(
                 CONF_BRIGHTNESS_SCALE, default=DEFAULT_BRIGHTNESS_SCALE
             ): vol.All(vol.Coerce(int), vol.Range(min=1)),
+            vol.Optional(
+                CONF_BRIGHTNESS_SCALE_MIN, default=DEFAULT_BRIGHTNESS_SCALE_MIN
+            ): vol.All(vol.Coerce(int), vol.Range(min=1)),
```

Comment:
I do not like this option, also it should not be in this PR. If you want to propose this, do this in a separate PR.